controllers/mattermost/mattermost/mattermost.go

```diff
@@ -85,116 +114,169 @@ func (r *MattermostReconciler) checkMattermostService(
+		return recStatus
 	}
 
 	current := &rbacv1.Role{}
 	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, current)
 	if err != nil {
-		return err
+		recStatus.Error = err
+		return recStatus
 	}
 
-	return r.Resources.Update(current, desired, reqLogger)
+	recStatus.Error = r.Resources.Update(current, desired, reqLogger)
+
+	return recStatus
 }
 
-func (r *MattermostReconciler) checkMattermostRoleBinding(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) error {
+func (r *MattermostReconciler) checkMattermostRoleBinding(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) reconcileStatus {
+	recStatus := reconcileStatus{
+		Status: false,
+		Error:  nil,
+	}
+
 	desired := mattermostApp.GenerateRoleBindingV1Beta(mattermost, mattermost.Name, mattermost.Name)
 	err := r.Resources.CreateRoleBindingIfNotExists(mattermost, desired, reqLogger)
 	if err != nil {
-		return err
+		recStatus.Error = err
+		return recStatus
 	}
 
 	current := &rbacv1.RoleBinding{}
 	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, current)
 	if err != nil {
-		return err
+		recStatus.Error = err
+		return recStatus
 	}
 
-	return r.Resources.Update(current, desired, reqLogger)
+	recStatus.Error = r.Resources.Update(current, desired, reqLogger)
+	return recStatus
 }
 
-func (r *MattermostReconciler) checkMattermostIngress(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) error {
+func (r *MattermostReconciler) checkMattermostIngress(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) reconcileStatus {
+	recStatus := reconcileStatus{
+		Status: false,
+		Error:  nil,
+	}
+
 	desired := mattermostApp.GenerateIngressV1Beta(mattermost)
 
 	if !mattermost.IngressEnabled() {
 		err := r.Resources.DeleteIngress(types.NamespacedName{Namespace: desired.Namespace, Name: desired.Name}, reqLogger)
 		if err != nil {
-			return errors.Wrap(err, "failed to delete disabled ingress")
+			recStatus.Error = errors.Wrap(err, "failed to delete disabled ingress")
+			return recStatus
 		}
-		return nil
+		return recStatus
 	}
 
 	err := r.Resources.CreateIngressIfNotExists(mattermost, desired, reqLogger)
 	if err != nil {
-		return err
+		recStatus.Error = err
+		return recStatus
 	}
 
 	current := &networkingv1.Ingress{}
 	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, current)
 	if err != nil {
-		return err
+		recStatus.Error = err
+		return recStatus
 	}
 
-	return r.Resources.Update(current, desired, reqLogger)
+	recStatus.Error = r.Resources.Update(current, desired, reqLogger)
+
+	return recStatus
 }
 
 func (r *MattermostReconciler) checkMattermostDeployment(
 	mattermost *mmv1beta.Mattermost,
 	dbConfig mattermostApp.DatabaseConfig,
 	fileStoreInfo *mattermostApp.FileStoreInfo,
 	status *mmv1beta.MattermostStatus,
-	reqLogger logr.Logger) error {
+	reqLogger logr.Logger) reconcileStatus {
```

Comment:
Let's keep the error here separate, as I mentioned in the above comment:
```suggestion
	reqLogger logr.Logger) (reconcileStatus, error) {
```